modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportReindexAction.java

```diff
@@ -315,15 +321,31 @@ void finishHim(Exception failure, List<Failure> indexingFailures, List<SearchFai
 
             // id and source always come from the found doc. Scripts can change them but they operate on the index request.
             index.id(doc.getId());
-            index.source(doc.getSource());
+
+            // the source xcontent type and destination could be different
```

Comment:
why would it be? can't we just ask for the same content-type when retrieving docs using _search ? Or is there something I am missing here? e.g. if you store yaml you still get it back as json when calling search without Accept header.